### ISSUE TITLE
Implement CNN model for CIFAR-10 classification using TensorFlow/Keras

### ISSUE THREAD

**Issue by HERRSAN** [HERRSAN/netlify-feature-tour#1]
### Overview
Implement a convolutional neural network (CNN) for image classification on the CIFAR-10 dataset using TensorFlow and Keras. The model should:
- Load and normalize CIFAR-10 data
- Build a sequential CNN architecture
- Compile and train the network
- Evaluate performance on the validation set

### Steps
1. **Load Real CIFAR-10 Data**
    - Use `cifar10.load_data()` to obtain and split training/validation images and labels.
    - Normalize pixel values to [0.0, 1.0].
    - Inspect shapes and data consistency.

2. **Define Model Architecture**
    - Setup input layer for shape (32, 32, 3).
    - Add two Conv2D + MaxPooling2D combinations.
    - Flatten and add a dense layer.
    - Output layer: 10 neurons, softmax activation for one-hot classes.

3. **Compile Model**
    - Use Adam optimizer.
    - Loss function: `sparse_categorical_crossentropy` (for integer labels).
    - Metric: Accuracy.

4. **Train Model**
    - Use ~10 epochs, batch size 64.
    - Track training/validation accuracy and loss.

5. **Evaluate Model**
    - Run `model.evaluate()` on validation data.
    - Report validation loss and accuracy.

### Reference Code
```python
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras.datasets import cifar10

# --- 1. CARGAR DATOS REALES (CIFAR-10) ---
(x_entrenamiento, y_entrenamiento), (x_validacion, y_validacion) = cifar10.load_data()
x_entrenamiento = x_entrenamiento.astype("float32") / 255.0
x_validacion = x_validacion.astype("float32") / 255.0
IMG_HEIGHT = x_entrenamiento.shape[1]
IMG_WIDTH = x_entrenamiento.shape[2]
IMG_CHANNELS = x_entrenamiento.shape[3]
NUM_CLASES = 10

# --- 2. DEFINIR LA ARQUITECTURA DEL MODELO ---
model = keras.Sequential([
    layers.Conv2D(32, (3, 3), activation='relu', input_shape=(IMG_HEIGHT, IMG_WIDTH, IMG_CHANNELS)),
    layers.MaxPooling2D((2, 2)),
    layers.Conv2D(64, (3, 3), activation='relu'),
    layers.MaxPooling2D((2, 2)),
    layers.Flatten(),
    layers.Dense(64, activation='relu'),
    layers.Dense(NUM_CLASES, activation='softmax')
])
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])

# --- 4. ENTRENAR EL MODELO ---
history = model.fit(x_entrenamiento, y_entrenamiento, epochs=10, batch_size=64, validation_data=(x_validacion, y_validacion), verbose=1)

# --- 5. EVALUAR EL MODELO ---
test_loss, test_acc = model.evaluate(x_validacion, y_validacion, verbose=0)
print(f"Pérdida de validación: {test_loss:.4f}")
print(f"Precisión de validación: {test_acc:.4f}")
```

### Acceptance Criteria
- Model achieves reasonable validation accuracy (>60% typical for simple CNN)
- Training/validation loss and accuracy outputs are reported and reproducible
- All scripts and code should be Python 3 with TensorFlow and Keras, and include necessary comments/documentation

**Comment by HERRSAN** [comment]
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras.datasets import cifar10

# --- 1. CARGAR DATOS REALES (CIFAR-10) ---
(x_entrenamiento, y_entrenamiento), (x_validacion, y_validacion) = cifar10.load_data()
x_entrenamiento = x_entrenamiento.astype("float32") / 255.0
x_validacion = x_validacion.astype("float32") / 255.0
IMG_HEIGHT = x_entrenamiento.shape[1]
IMG_WIDTH = x_entrenamiento.shape[2]
IMG_CHANNELS = x_entrenamiento.shape[3]
NUM_CLASES = 10

# --- 2. DEFINIR LA ARQUITECTURA DEL MODELO ---
model = keras.Sequential([
    layers.Conv2D(32, (3, 3), activation='relu', input_shape=(IMG_HEIGHT, IMG_WIDTH, IMG_CHANNELS)),
    layers.MaxPooling2D((2, 2)),
    layers.Conv2D(64, (3, 3), activation='relu'),
    layers.MaxPooling2D((2, 2)),
    layers.Flatten(),
    layers.Dense(64, activation='relu'),
    layers.Dense(NUM_CLASES, activation='softmax')
])
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])

# --- 4. ENTRENAR EL MODELO ---
history = model.fit(x_entrenamiento, y_entrenamiento, epochs=10, batch_size=64, validation_data=(x_validacion, y_validacion), verbose=1)

# --- 5. EVALUAR EL MODELO ---
test_loss, test_acc = model.evaluate(x_validacion, y_validacion, verbose=0)
print(f"Pérdida de validación: {test_loss:.4f}")
print(f"Precisión de validación: {test_acc:.4f}")
